### PR TITLE
Feed Parser

### PR DESCRIPTION
**Pull request by olaydub**
Feed parser to parse XML

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,15 +29,17 @@
         "url": "https://github.com/wandyezj/website/issues"
     },
     "homepage": "https://github.com/wandyezj/website#readme",
-    "dependencies": {},
+    "dependencies": {
```
Comment (wandyezj) [inline package.json]: Why is this dependency needed?

File: package.json [hunk 2]
```diff
@@ -29,15 +29,17 @@
...
+    },
     "devDependencies": {
         "@playwright/test": "^1.32.3",
         "@types/node": "^18.7.17",
-        "playwright": "^1.33.0",
-        "prettier": "^2.8.8",
         "clean-webpack-plugin": "^4.0.0",
         "copy-webpack-plugin": "^7.0.0",
         "html-webpack-plugin": "^5.5.0",
+        "playwright": "^1.33.0",
```
Comment (wandyezj) [inline package.json]: Why did these dev dependencies move?

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const express = require('express');
```
Comment (wandyezj) [inline src/index.js]: You shouldn't need this index.js file.

File: src/parser.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+//const RSS_URL = 'https://feeds.npr.org/1001/rss.xml';
+
+
+//retrieve RSS feed from specified URL, processes response, logs parsed XML
+//data to console
+export function fetchAndParseRSS(RSS_URL : string) {
+    fetch(RSS_URL)
```
Comment (wandyezj) [inline src/parser.ts]: It's a little easier to use `async` `await` instead of then chains. see [async and await on MDN](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Promises#async_and_await)

File: webpack.config.js [hunk 5]
```diff
@@ -54,10 +54,26 @@ module.exports = async (env, options) => {
...
+                        to: "./test-xml/feeds.npr.org_1001_rss.xml",
+                        from: "./test-xml/feeds.npr.org_1001_rss.xml",
+                    },
                     {
                         to: "assets/logo.svg",
                         from: "./logo/logo.svg",
                     },
+                    {
+                        to: "index.ts",
+                        from: "./src/index.ts",
```
Comment (wandyezj) [inline webpack.config.js]: Why is this copied?

File: src/parser.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
+//const RSS_URL = 'https://feeds.npr.org/1001/rss.xml';
+
+//retrieve RSS feed from specified URL, processes response, logs parsed XML
+//data to console
+export function fetchAndParseRSS(RSS_URL: string) {
```
Comment (wandyezj) [inline src/parser.ts]: `fetchAndParseRSS` should have a unit test

File: .idea/workspace.xml [hunk 7]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (wandyezj) [inline .idea/workspace.xml]: Can we do this in a separate PR?

**Review by wandyezj — APPROVED** [review]

**Review by Jquin997 — APPROVED** [review]
